.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.6.4
+    hooks:
+      - id: ruff
+        args: [--fix]

```

Comment:
I compared with pybind11/.pre-commit-config.yaml, there we have:
```
    args: ["--fix", "--show-fixes"]                                             
```
Seeing the fixes might be helpful.

Reply:
agree, will add

Reply:
wrt to tox, I also agree. Actually I meant to put those changes in an experimental branch to see if I could avoid running the tests locally. However the solution to gh-action testing is not so simple and will require something more involved. Tox changes to be removed imminently.